### DISCUSSION TITLE
data sprintrace

### DISCUSSION THREAD

**Discussion by walteij** [theOehrly/Fast-F1#160] [General]
Anyone able to get some data from the sprintrace?

`core           INFO     Loading data for Emilia Romagna Grand Prix - Sprint Qualifying [v2.2.2]
api            INFO     No cached data found for driver_info. Loading data...
api            INFO     Fetching driver list...
core        WARNING     Failed to load extended driver information!
core        WARNING     Failed to load data from Ergast API! (This is expected for recent sessions)
api            INFO     No cached data found for timing_data. Loading data...
api            INFO     Fetching timing data...
core        WARNING     Failed to load lap data!
api            INFO     No cached data found for car_data. Loading data...
api            INFO     Fetching car data...
core        WARNING     Failed to load telemetry data!
api            INFO     No cached data found for weather_data. Loading data...
api            INFO     Fetching weather data...
core        WARNING     Failed to load weather data!
core           INFO     Finished loading data for 0 drivers: []`

**Comment by RobertoTonini** [comment]
I guess this is due to the fact that from 2022 onward it’s called “Sprint” and not “Sprint qualifying” anymore, so the data path on F1’s servers has changed. The api should be updated

**Comment by RobertoTonini** [comment]
To temporarily solve the issue I managed to find a simple solution by editing the api code. You have to add the following code to the events.py file:
add to the "_SESSION_TYPE_ABBREVIATIONS" dictionary the following element:
'S': 'Sprint'

Add to the "_get_schedule" function the following code:
df = pd.read_json(response.text)
if year == 2022:
df['session4'] = df['session4'].replace("Sprint Qualifying", "Sprint")

**Reply by walteij** [reply]
Yup, that did the trick

**Comment by theOehrly** [comment]
API consistency... oh well. Sorry, this is one of those things that I can't know in advance.
It's fixed with the new release just now. We now have the session types 'Sprint'/'Sprint Qualifying' and the abbreviations 'S'/'SQ'. You can use whichever you like, and FastF1 will silently correct the session name depending on the season.

**Comment by GeorgeC79** [comment]
Still getting the error after upgrading to 2.2.4. 

Traceback (most recent call last):
  File "C:\Python\Python38\lib\site-packages\fastf1\events.py", line 728, in get_session_name
    num = float(identifier)
ValueError: could not convert string to float: 'Sprint Qualifying'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python\Python38\lib\site-packages\fastf1\events.py", line 280, in get_session
    return event.get_session(identifier)
  File "C:\Python\Python38\lib\site-packages\fastf1\events.py", line 819, in get_session
    return Session(event=self, session_name=session_name,
  File "C:\Python\Python38\lib\site-packages\fastf1\core.py", line 887, in __init__
    self.date = self.event.get_session_date(session_name)
  File "C:\Python\Python38\lib\site-packages\fastf1\events.py", line 777, in get_session_date
    session_name = self.get_session_name(identifier)
  File "C:\Python\Python38\lib\site-packages\fastf1\events.py", line 749, in get_session_name
    raise ValueError(f"No session of type '{identifier}' for "
ValueError: No session of type 'Sprint Qualifying' for this event
>>> session.load_laps()

**Reply by theOehrly** [reply]
Ahh sorry, now it's working again. The bot that does the schedule updates messed it up.

**Reply by GeorgeC79** [reply]
Thanks

**Comment by walteij** [comment]
On another note. Some data is not visible when getting the results of the sprint race. Q1, Q2 and Q3 are obviously logical, but I am missing values for: Position, Gridpositions, Time, Status and Points:

`
core           INFO     Loading data for Emilia Romagna Grand Prix - Sprint [v2.2.4]
api            INFO     Using cached data for driver_info
api            INFO     Using cached data for timing_data
api            INFO     Using cached data for timing_app_data
core           INFO     Processing timing data...
api            INFO     Using cached data for session_status_data
core        WARNING     No lap data for driver 24
api            INFO     Using cached data for track_status_data
api            INFO     Using cached data for car_data
api            INFO     Using cached data for position_data
api            INFO     Using cached data for weather_data
core           INFO     Finished loading data for 20 drivers: ['1', '6', '10', '22', '18', '24', '44', '47', '63', '55', '5', '77', '11', '3', '14', '20', '4', '16', '31', '23']
>>> sprint.load()
core           INFO     Loading data for Emilia Romagna Grand Prix - Race [v2.2.4]
api            INFO     Using cached data for driver_info
api            INFO     Using cached data for timing_data
api            INFO     Using cached data for timing_app_data
core           INFO     Processing timing data...
api            INFO     Using cached data for session_status_data
api            INFO     Using cached data for track_status_data
api            INFO     Using cached data for car_data
api            INFO     Using cached data for position_data
api            INFO     Using cached data for weather_data
core           INFO     Finished loading data for 20 drivers: ['1', '11', '4', '63', '77', '16', '22', '5', '20', '18', '23', '10', '44', '31', '24', '6', '47', '3', '14', '55']
>>> race.results
   DriverNumber BroadcastName Abbreviation         TeamName TeamColor  FirstName    LastName          FullName  Position  GridPosition  Q1  Q2  Q3 Time Status  Points
1             1  M VERSTAPPEN          VER  Red Bull Racing    1e5bc6        Max  Verstappen    Max Verstappen       0.0           0.0 NaT NaT NaT  NaT            0.0
6             6      N LATIFI          LAT         Williams    37bedd   Nicholas      Latifi   Nicholas Latifi       0.0           0.0 NaT NaT NaT  NaT            0.0
10           10       P GASLY          GAS       AlphaTauri    4e7c9b     Pierre       Gasly      Pierre Gasly       0.0           0.0 NaT NaT NaT  NaT            0.0
22           22     Y TSUNODA          TSU       AlphaTauri    4e7c9b       Yuki     Tsunoda      Yuki Tsunoda       0.0           0.0 NaT NaT NaT  NaT            0.0
18           18      L STROLL          STR     Aston Martin    2d826d      Lance      Stroll      Lance Stroll       0.0           0.0 NaT NaT NaT  NaT            0.0
24           24        G ZHOU          ZHO       Alfa Romeo    b12039     Guanyu        Zhou       Guanyu Zhou       0.0           0.0 NaT NaT NaT  NaT            0.0
44           44    L HAMILTON          HAM         Mercedes    6cd3bf      Lewis    Hamilton    Lewis Hamilton       0.0           0.0 NaT NaT NaT  NaT            0.0
47           47  M SCHUMACHER          MSC     Haas F1 Team    b6babd       Mick  Schumacher   Mick Schumacher       0.0           0.0 NaT NaT NaT  NaT            0.0
63           63     G RUSSELL          RUS         Mercedes    6cd3bf     George     Russell    George Russell       0.0           0.0 NaT NaT NaT  NaT            0.0
55           55       C SAINZ          SAI          Ferrari    ed1c24     Carlos       Sainz      Carlos Sainz       0.0           0.0 NaT NaT NaT  NaT            0.0
5             5      S VETTEL          VET     Aston Martin    2d826d  Sebastian      Vettel  Sebastian Vettel       0.0           0.0 NaT NaT NaT  NaT            0.0
77           77      V BOTTAS          BOT       Alfa Romeo    b12039   Valtteri      Bottas   Valtteri Bottas       0.0           0.0 NaT NaT NaT  NaT            0.0
11           11       S PEREZ          PER  Red Bull Racing    1e5bc6     Sergio       Perez      Sergio Perez       0.0           0.0 NaT NaT NaT  NaT            0.0
3             3   D RICCIARDO          RIC          McLaren    f58020     Daniel   Ricciardo  Daniel Ricciardo       0.0           0.0 NaT NaT NaT  NaT            0.0
14           14      F ALONSO          ALO           Alpine    2293d1   Fernando      Alonso   Fernando Alonso       0.0           0.0 NaT NaT NaT  NaT            0.0
20           20   K MAGNUSSEN          MAG     Haas F1 Team    b6babd      Kevin   Magnussen   Kevin Magnussen       0.0           0.0 NaT NaT NaT  NaT            0.0
4             4      L NORRIS          NOR          McLaren    f58020      Lando      Norris      Lando Norris       0.0           0.0 NaT NaT NaT  NaT            0.0
16           16     C LECLERC          LEC          Ferrari    ed1c24    Charles     Leclerc   Charles Leclerc       0.0           0.0 NaT NaT NaT  NaT            0.0
31           31        E OCON          OCO           Alpine    2293d1    Esteban        Ocon      Esteban Ocon       0.0           0.0 NaT NaT NaT  NaT            0.0
23           23       A ALBON          ALB         Williams    37bedd  Alexander       Albon   Alexander Albon       0.0           0.0 NaT NaT NaT  NaT            0.0`

**Reply by theOehrly** [reply]
Similar problem with 'Sprint Qualifying' being called 'Sprint' now. 

Also, feel free to report something like this as a bug.  While it doesn't matter too much, it's good to keep questions and general discussion separated from bug reports. It's a bit easier to keep track of everything then.

**Reply by walteij** [reply]
Done

**Comment by NOCTish** [comment]
Hello There!
I've recently found this code (which I'm using last version RN) and have wrote this so far:
```
import fastf1 as ff1
import os
script_dir = os.path.dirname(os.path.abspath(__file__))
cache_dir = os.path.join(script_dir, 'cache')
if not os.path.exists(cache_dir):
    os.makedirs(cache_dir)
ff1.Cache.enable_cache(cache_dir)

YEAR = 2022
RACE = 'Bahrain'
SESSION = 'R'

session = ff1.get_session(
    YEAR,
    RACE,
    SESSION
)

session.load()
laps = session.laps
laps.pick_fastest()
laps.pick_driver('HAM')

alo_laps = laps.pick_driver('ALO')
alo_fastest = alo_laps.pick_fastest()

alo_fastest.get_car_data()
```

But unfortunetly I get these errors and have no idea what to do:
PS C:\Users\amirm> & C:/Users/amirm/AppData/Local/Programs/Python/Python310/python.exe "c:/Users/amirm/Desktop/F1/import fastf1 as ff1.py"                                                               core           INFO     Loading data for Bahrain Grand Prix - Race [v2.3.0]
api            INFO     No cached data found for driver_info. Loading data...
api            INFO     Fetching driver list...
core        WARNING     Failed to load extended driver information!
api            INFO     No cached data found for timing_data. Loading data...
api            INFO     Fetching timing data...
core        WARNING     Failed to load lap data!
api            INFO     No cached data found for car_data. Loading data...
api            INFO     Fetching car data...
core        WARNING     Failed to load telemetry data!
api            INFO     No cached data found for weather_data. Loading data...
api            INFO     Fetching weather data...
core        WARNING     Failed to load weather data!
api            INFO     No cached data found for race_control_messages. Loading data...
api            INFO     Fetching race control messages...
core        WARNING     Failed to load Race Control message data!
core           INFO     Finished loading data for 20 drivers: ['16', '55', '44', '63', '20', '77', '31', '22', '14', '24', '47', '18', '23', '3', '4', '6', '27', '11', '1', '10']
Traceback (most recent call last):
  File "c:\Users\amirm\Desktop\F1\import fastf1 as ff1.py", line 20, in <module>
    laps = session.laps
  File "C:\Users\amirm\AppData\Local\Programs\Python\Python310\lib\site-packages\fastf1\core.py", line 1052, in laps
    return self._get_property_warn_not_loaded('_laps')
  File "C:\Users\amirm\AppData\Local\Programs\Python\Python310\lib\site-packages\fastf1\core.py", line 1017, in _get_property_warn_not_loaded
    raise DataNotLoadedError("The data you are trying to access has not "
fastf1.core.DataNotLoadedError: The data you are trying to access has not been loaded yet. See `Session.load`
